src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/simple/RandomizedJoinPlanOptimizerImpl.java

```diff
@@ -0,0 +1,50 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.simple;
+
+import java.util.List;
+import java.util.Random;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+
+public class RandomizedJoinPlanOptimizerImpl implements JoinPlanOptimizer {
+	
+	/**
+	 * The class contains a random seed object.
+	 * This cannot be marked final (compilation error if so).
+	 */
+	protected Random random = new Random();
+	
+	/**
+	 * Have a list of subplans.
+	 */
+	protected final List<PhysicalPlan> subplans;
+	
+	/**
+	 *  When initializing the algorithm, start with the given list of subplans.
+	 *  @param subplans
+	 */
+	public RandomizedJoinPlanOptimizerImpl( final List<PhysicalPlan> subplans ) {
+		this.subplans = subplans;
+		this.random = new Random(); // Generates the random seed.

```

Comment:
```suggestion
```

That's the part that is not needed because you are already assigning this member variable in its declaration.